cpp/src/table/row_operators.cu

```diff
@@ -77,50 +149,75 @@ auto decompose_structs(table_view table,
                        host_span<order const> column_order         = {},
                        host_span<null_order const> null_precedence = {})
 {
+  auto sliced         = pushdown_struct_offsets(table);
+  auto linked_columns = detail::table_to_linked_columns(sliced);
+
   std::vector<column_view> verticalized_columns;
   std::vector<order> new_column_order;
   std::vector<null_order> new_null_precedence;
   std::vector<int> verticalized_col_depths;
-  for (size_type col_idx = 0; col_idx < table.num_columns(); ++col_idx) {
-    auto const& col = table.column(col_idx);
-    if (is_nested(col.type())) {
+  for (size_t col_idx = 0; col_idx < linked_columns.size(); ++col_idx) {
+    auto const col = linked_columns[col_idx];
```

Comment:
Could this use a range-based for loop instead?

Reply:
Need the index later to index into column_order